common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -342,16 +360,40 @@ def from_storable(self, edit_info):
         self.original_usage = edit_info.get('original_usage', None)
         self.original_usage_version = edit_info.get('original_usage_version', None)
 
-    def __str__(self):
-        return ("EditInfo(previous_version={0.previous_version}, "
-                "update_version={0.update_version}, "
-                "source_version={0.source_version}, "
-                "edited_on={0.edited_on}, "
-                "edited_by={0.edited_by}, "
-                "original_usage={0.original_usage}, "
-                "original_usage_version={0.original_usage_version}, "
-                "_subtree_edited_on={0._subtree_edited_on}, "
-                "_subtree_edited_by={0._subtree_edited_by})").format(self)
+    def get_source_version(self, value_if_unset=UNSET):
+        """
+        Return the version of the given database representation of a block.
+        """
+        if value_if_unset is UNSET:
+            # Parameter was not passed-in.
+            value_if_unset = self.update_version
+        if self.source_version is not UNSET:
+            return self.source_version
+        else:
+            return value_if_unset
+
+    def __repr__(self):
+        # pylint: disable=bad-continuation
+        return ("{classname}(previous_version={previous_version}, "
+                "update_version={update_version}, "
+                "source_version={source_version}, "
+                "edited_on={edited_on}, "
+                "edited_by={edited_by}, "
+                "original_usage={original_usage}, "
+                "original_usage_version={original_usage_version}, "
+                "_subtree_edited_on={subtree_edited_on}, "
+                "_subtree_edited_by={subtree_edited_by})").format(
+            classname=self.__class__.__name__,
+            previous_version=self.previous_version,
+            update_version=self.update_version,
+            source_version="UNSET" if self.source_version is UNSET else self.source_version,
+            edited_on=self.edited_on,
+            edited_by=self.edited_by,
+            original_usage=self.original_usage,
+            original_usage_version=self.original_usage_version,
+            subtree_edited_on=self._subtree_edited_on,
+            subtree_edited_by=self._subtree_edited_by,
+        )  # pylint: disable=bad-continuation
```

Comment:
Nit: I think it was actually more readable when you passed `self` in, and accessed the attributes in the format strings. I just thought that rather than using `{0.some_attribute}`, `{self.some_attribute}` would be clearer.